typescript/approvals.d.ts

```diff
@@ -189,16 +189,32 @@ export type FormApprovalWebhookEvent =
   | 'NOTE_UPDATED'
   | 'NOTE_DELETED'
 
-export type NewFormApprovalWebhook = {
+export type BaseFormApprovalWebhook = {
   endpoint: EndpointConfiguration
-  secret: string
   webhookEvents: FormApprovalWebhookEvent[]
   isEnabled: boolean
   formsAppId: number
   formIds?: number[]
   label: string
 }
 
+export type FormApprovalWebhookCreatePayload = BaseFormApprovalWebhook & {
+  /** Must pass a `secret` to create a webhook */
+  secret: string
+}
+
+export type FormApprovalWebhookUpdatePayload = BaseFormApprovalWebhook & {
+  /**
+   * `secret` is optional when updating a webhook. If no secret is passed, the
+   * secret will remain unchanged.
+   */
+  secret?: string
+}
+
+export type NewFormApprovalWebhook = BaseFormApprovalWebhook & {
+  customerSecretId: number
+}
```

Comment:
This seems really confusing to me. Why does a `NewFormApprovalWebhook` require an existing secretId?

Reply:
`NewFormApprovalWebhook` is the type passed to our database layer. It requires everything that is needed to create a `FormApprovalWebhook` from the database layer's perspective. That is the pattern the majority (potentially all) of our `New*` types follow.

Reply:
Hm. For most it is also the type the client needs to pass to the api though.

Reply:
I understand it seems obvious to you, but you are one person who thinks about it in one way.

Reply:
Happy coincidence. I'm happy to rename the type: `FormApprovalWebhookDatabaseCreatePayload`?